crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -515,14 +529,92 @@ fn create_get_pending_operations_message() -> Result<Message, ConversionError> {
 fn create_supported_operations_fragments_message(
     cfg_dir: &Path,
 ) -> Result<Message, ConversionError> {
+    let dir_split: Vec<&str> = cfg_dir
+        .to_str()
+        .ok_or_else(|| ConversionError::DirPathComponentError {
+            dir: cfg_dir.to_path_buf(),
+        })?
+        .split('/')
+        .collect();
+
+    if dir_split.len() == 6 {
+        let stopic = format!("{SMARTREST_PUBLISH_TOPIC}/{}", dir_split[5]);
```

Comment:
This logic to extract the child id from a path is copy-pasted many time. Please add a function.